src/diffpy/labpdfproc/tools.py

```
@@ -133,21 +140,21 @@ def set_wavelength(args):
     args : argparse.Namespace
         The updated arguments with the wavelength.
     """
-    if args.wavelength is not None and args.wavelength <= 0:
-        raise ValueError(
-            "No valid wavelength. Please rerun specifying a known anode_type or a positive wavelength."
-        )
-    if not args.wavelength and args.anode_type and args.anode_type not in WAVELENGTHS:
-        raise ValueError(
-            f"Anode type not recognized. Please rerun specifying an anode_type from {*known_sources, }."
-        )
-
-    if args.wavelength:
-        delattr(args, "anode_type")
-    elif args.anode_type:
+    if args.wavelength is None:
+        matched_anode_type = next((key for key in WAVELENGTHS if key.lower() == args.anode_type.lower()), None)
+        if matched_anode_type is None:
+            raise ValueError(
+                f"Anode type not recognized. Please rerun specifying an anode_type from {*known_sources, }."
+            )
+        args.anode_type = matched_anode_type
         args.wavelength = WAVELENGTHS[args.anode_type]
     else:
-        args.wavelength = WAVELENGTHS["Mo"]
+        if args.wavelength <= 0:
+            raise ValueError(
+                "No valid wavelength. Please rerun specifying a known anode_type or a positive wavelength."
+            )
+        else:
+            delattr(args, "anode_type")
```

do I understand that if the user enters both a valid wavelength and an anode type we default to the wavelength?  Is there a good use-case where this makes sense?  Otherwise it looks to me like magic and we should just crash if the user provides both (i.e., I can see that they would do htat by mistake, I can't see that they would do that on purpose, though there may be a UC that you have in mind.

---

Yes so if the user enters only wavelength=0.25 and leaves the anode type empty, it will be automatically loaded as anode type = Mo, but that's wrong. Maybe we can add a check box somewhere to allow user to remove anode type (or automatically remove it when user enters a wavelength, which is kind of what we're doing now)?

---

I would throw an error if both are specified.  Unless you can think of a case where the user would intentionally do that for some reason.